### DISCUSSION TITLE
DS1820 Rule on Tasmota 12.2.0

### DISCUSSION THREAD

**Discussion by storfusker** [arendst/Tasmota#17098] [Q&A] [answered]
Hi guys 

What am I missing here

Have this "Simple" rule on a Wemos D1 mini where I have one (Or more) DS18B20 temperature sensors connected

The temperatures from the connected DS18B20 devices are displaying correctly in the Main window

I can toggle Relay / LED connected to a gpio pin and configures as Relay1 by issuing the command
POWER1 on or POWER1 1 and off again by POWER1 off or POWER1 0

So far so good :-)

But I cant make the rule working  that is supposed to switch on the Relay or LED on when the temperature from one of the DS18B20 exceeds a certain temperature

Output from Rule1 command:

3:57:55.014 CMD: Rule1
13:57:55.022 MQT: stat/tasmota_82ADF5/RESULT = {"Rule1":{"State":"OFF","Once":"OFF","StopOnError":"OFF","Length":48,"Free":463,"Rules":"ON DS1820-1#Temperature > 27.1 DO POWER1 1 ENDON"}}

Tried using different values for temperature and also tried integer values, so its not this parameter that is cheating me

Output from SetOption64 command (So the sensors are using hyphen in names);
13:59:05.848 CMD: SetOption64
13:59:05.854 MQT: stat/tasmota_82ADF5/RESULT = {"SetOption64":"OFF"}

What am I missing ?

**Comment by storfusker** [comment, marked as answer]
I was missing a "B"

Looking deeper into the names I discovered that the sensor is named DS18B20-1#Temperature and not DS1820-1#Temperature as I used in the rule
Maybe a check of sensor names would be useful int the code. The app did not complain about "My" wrong name in  any way

**Reply by arendst** [reply]
There was nothing wrong with "Your" name as long as you connect DS1820 instead of DS18B20....

https://www.futurlec.com/Dallas/DS1820.shtml

**Reply by storfusker** [reply]
My first Tasmota Rule so plenty of things to take into account :-)
Thanks for clarifying....it makes very much sense now
